Update flatpak runtime to Gnome 48

Vraiment opened the pull request:
I'm trying to fix #12 , from this:

> Looks like json-glib was dropped from the [upstream Flatpak manifest](https://gitlab.gnome.org/raggesilver/blackbox/-/commit/313b1667ea1cdb6005f004f6b57f64fca0d65f82), probably because it is included in the GNOME runtime

I gather that it doesn't require to build `json-glib` anymore because is included on the Gnome runtime so the build should not fail anymore.

Vraiment committed:
Update flatpak runtime to Gnome 48

Vraiment committed:
Add a patch just to test the build

prateektade commented:
[Marble has also been dropped](https://gitlab.gnome.org/raggesilver/blackbox/-/commit/2895855011a8bcbd2bff1d9c3677be76c6111ba2)

Vraiment committed:
Drop marble

https://gitlab.gnome.org/raggesilver/blackbox/-/commit/2895855011a8bcbd2bff1d9c3677be76c6111ba2

technobirb commented:
The errors stopping the build are actually these 2:
```
In file included from /usr/include/glib-2.0/glib.h:64:
../../../../../blackbox/src/widgets/ColorSchemeThumbnail.vala: In function ‘terminal_color_scheme_preview_paintable_load_image_co’:
/usr/include/glib-2.0/glib/gmessages.h:658:9: error: ‘return’ with no value, in function returning non-void [-Wreturn-mismatch]
  658 |         return; \
      |         ^~~~~~
../../../../../blackbox/src/widgets/ColorSchemeThumbnail.vala:128:2: note: in expansion of macro ‘g_return_if_fail’
  128 |     return_if_fail (file_content != null);
      |  ^  ~~~~~~~~~~~~~
ColorSchemeThumbnail.c:1234:1: note: declared here

In file included from /usr/include/glib-2.0/glib.h:64,
                 from /usr/include/glib-2.0/gobject/gbinding.h:30,
                 from /usr/include/glib-2.0/glib-object.h:24,
                 from src/blackbox.p/widgets/Window.c:23:
../../../../../blackbox/src/widgets/Window.vala: In function ‘terminal_padding_from_variant’:
../../../../../blackbox/src/widgets/Window.vala:41:79: error: ‘return’ with a value, in function returning void [-Wreturn-mismatch]
   41 |     return_val_if_fail (
/usr/include/glib-2.0/glib/gmessages.h:671:17: note: in definition of macro ‘g_return_val_if_fail’
  671 |         return (val); \
      |                 ^~~
Window.c:824:1: note: declared here
```
They are caused by Vala [not generating the correct C code](https://gitlab.gnome.org/GNOME/vala/-/issues/1005) when using those macros, and have been changed in this [commit](https://gitlab.gnome.org/raggesilver/blackbox/-/commit/d693b9289768c22c59a2f336ee49dc6057d1db93).
Though I'm not sure why it didn't happen with older versions, different Vala version in the GNOME SDK perhaps?

The build also warns about deprecated GTK widgets (fixed in main), and Idk if there are more fixes of the same nature. Maybe the proper way to update the runtime is to use the main branch...

ryonakano commented:
> Maybe the proper way to update the runtime is to use the main branch...

I think this is impossible on Flathub, they requires to use a fixed revision for reproducible builds. We can create a patch to fix the issue, place it in your branch, and apply it while building instead.

yonas commented:
What's the status of this? The flatpak is still out of date.

Vraiment commented:
I'm no longer working on this
